File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool isPrime(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
    
    bool halvesAreAlike(string s) {
        int count = 0, n = s.size();
        for (int i = 0; i < n; i++) {
            if (isPrime(tolower(s[i]))) {
                if (i < n / 2) {
                    count++;
                } else {
                    count--;
                }
            }
        }
        return count == 0;
    }
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public static String VOWELS = "aeiouAEIOU";
    
    public boolean halvesAreAlike(String s) {
        int count = 0;
        for (int i = 0; i < s.length() / 2; i++) {
            char ch = s.charAt(i);
            if (VOWELS.indexOf(ch) != -1) {
                count += 1;
            }
        }
        for (int i = s.length() / 2; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (VOWELS.indexOf(ch) != -1) {
                count -= 1;
            }
        }
        return count == 0;
    }
}